gpt_index/data_structs/data_structs_v2.py

```diff
@@ -0,0 +1,398 @@
+"""Data structures v2.
+
+Nodes are decoupled from the indices.
+
+"""
+
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, Sequence, Set, Tuple
+
+from dataclasses_json import DataClassJsonMixin
+
+from gpt_index.data_structs.data_structs import IndexStruct, Node
+from gpt_index.data_structs.struct_type import IndexStructType
+
+# OIS = TypeVar("IS", bound=IndexStruct)
+
+
+@dataclass
+class V2IndexStruct(IndexStruct, DataClassJsonMixin):
+    """A base data struct for a LlamaIndex."""
+
+    # # @abstractmethod
+    # @classmethod
+    # def from_v0_struct(cls, v0_struct: OIS) -> "V2IndexStruct":
+    #     """Convert from v0 struct."""
+    #     raise NotImplementedError()
```

Comment:
@jerryjliu Is this an attempt to maintain backwards compatibility? Is that possible? I suggest we just give up on backwards compatibility given the scale of the change.

Reply:
hmmm yeah maybe too much effort.